lib/graphql/schema/input_object.rb

```diff
@@ -171,7 +171,7 @@ def validate_non_null_input(input, ctx)
           # We're not actually _using_ the coerced result, we're just
           # using these methods to make sure that the object will
           # behave like a hash below, when we call `each` on it.
-          begin
+          input = begin
```

Comment:
This resolves https://github.com/rmosolgo/graphql-ruby/pull/2985#discussion_r439458182

Reply:
The comment on this block is now wrong

Reply:
Good catch! 🔥